app/models/spree/payment/processing.rb

```diff
@@ -15,6 +15,7 @@ def process!
 
       def process_offline!
         return unless validate!
+        return if cvv_response_message.present?
```

Comment:
I really think it's worth extracting a `Payment#needs_authorization?` out of this line. I know that's what this line of code means because you told me recently, but I'm sure other devs won't in the future, or even myself in a few months' time.

This, as a consequence, would make tests more readable because we could stub `needs_authorization?` instead of having to provide a string for the `cvv_response_message` attribute. It feels more intention-revealing to me.

Reply:
👍 9104ca72a9b52c94f506d200a5bfeace16280b8e